### PR TITLE
feat(unify): show reporter header; open and close specs list from it

### PR DESCRIPTION
**Pull request by mapsandapps**
* override global SVG styles ([1x56ga2](https://app.clickup.com/t/1x56ga2))
* expand and collapse specs list via reporter header ([1m9167h](https://app.clickup.com/t/1m9167h)) (paired with Mark)
* make reporter header visible when running in unified UI

Closes https://cypress-io.atlassian.net/browse/UNIFY-551

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/src/runner/SpecRunner.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@
       <InlineSpecList :gql="props.gql" />
     </HideDuringScreenshot>
 
-    <HideDuringScreenshot class="w-128">
+    <HideDuringScreenshot class="min-w-320px">
```
Comment (marktnoonan) [inline packages/app/src/runner/SpecRunner.vue:13]: (Just noting that this is a temporary measure until drag to resize gets implemented)

**Review by marktnoonan — COMMENTED** [review]
LGTM :)

File: packages/reporter/cypress/integration/header_spec.ts [hunk 2]
```diff
@@ -31,30 +31,6 @@ describe('header', () => {
     })
   })
 
-  describe('tests button', () => {
```
Comment (mapsandapps) [inline packages/reporter/cypress/integration/header_spec.ts:34]: Ideally, these tests should move somewhere, but I couldn't find any tests that display both the Vue & React portions of this. When only the React parts get displayed (like in these tests), the button doesn't get displayed at all. If anybody has an idea for where to put them, i can move/rewrite them.

Comment (marktnoonan) [reply packages/reporter/cypress/integration/header_spec.ts:34]: I was thinking about this this morning actually, I think our new e2e tests for the app should be able to show this, will double check.

Comment (marktnoonan) [reply packages/reporter/cypress/integration/header_spec.ts:34]: confirmed we don't have the ability to do this in e2e test atm, at least not easily. we can add a vue CT test though that renders the portal and makes sure the Vue part works correctly.

Comment (lmiller1990) [reply packages/reporter/cypress/integration/header_spec.ts:34]: We need to work on supporting E2E testing for that page...

File: packages/app/src/specs/InlineSpecList.vue [hunk 3]
```diff
@@ -87,4 +113,49 @@ const specs = computed(() => {
...
 })
 
+const renderTeleport = ref(false)
+const showList = ref(true)
+let teleportMutuationObserver: MutationObserver | null = null
+
+onKeyStroke('f', () => {
+  showList.value = !showList.value
+})
+
```
Comment (lmiller1990) [inline packages/app/src/specs/InlineSpecList.vue:123]: This seems like a very convoluted way of doing this. Why are we teleporting this button from Vue to React at all? I don't think this makes a ton of sense - all the checks, mutation observer and usage of `nextTick` are a bit of code smell to me.

I also don't think we should mix React and Vue like this where possible.

We already have a pattern for communicating between different parts of the runner UI (runner, reporter, driver), which is the `eventManager`.

What I'd suggest:

- In reporter, create the button using React. When you click it, you will do `events.emit('hide:specs')`, like [here](https://github.com/cypress-io/cypress/blob/10.0-release/packages/reporter/src/header/header.tsx#L24) for example.

- In the reporter events, https://github.com/cypress-io/cypress/blob/10.0-release/packages/reporter/src/lib/events.ts#L241, add something like

```ts
 localBus.on('hide:specs', (cb) => {
      runner.emit('hide:specs', cb)
    })
```

Confusing, `runner` refers to`eventManager` here, I'm fairly sure.

Now listen for that event in `eventManger`, here: https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/src/runner/event-manager.ts#L200 create something like

```ts
reporterBus.on('hide:specs', () => {
  this.emit('hide:specs')
})
```

Finally, then listen for that event in `SpecRunner.vue`, like this: https://github.com/cypress-io/cypress/blob/10.0-release/packages/app/src/runner/SpecRunner.vue#L116-L118

```ts
const runnerStore = useRunnerStore()

eventManager.on('hide:specs', () => {
  runnerStore.toggleSpecList()
})
```

Note that `runnerStore` doesn't exist, but I think we want to create a new store for all the runner UI state - there's a lot more coming. The stores are here: https://github.com/cypress-io/cypress/tree/10.0-release/packages/app/src/store

**Review by lmiller1990 — CHANGES_REQUESTED** [review]
Hi! I left a comment https://github.com/cypress-io/cypress/pull/18921/files#r752208170 which is fairly specific. We could pair on this - I think the current approach, while it seems to work, is quite complex and not ideal. I would really prefer to stick with the event manager pattern we've got here - it's more consistent. I am not super comfortable with teleporting Vue into React, I don't think we should be coupling these two parts of the interface like this.

**Comment by mapsandapps** [comment]
Closing in favor of #18991

**Pull request closed without merging**